### PR TITLE
timestamps again

### PR DESCRIPTION
**Pull request by chrisnojima**
tldr; This removes the timestamps entirely from the store and instead does it during rendering time

----------------------------------------------------

Ok, so
I wanted to investigate this as I was worried this was a list-rendering problem with the recent react-virtualized changes. It wasn't but I was thinking i'd spend < 1 hr on changing how this works.

I _think_ this is simpler and better.

Instead of injecting this (in 3 places) into the store, we instead just defer this until we draw. The Message wrappers already get the previous message to do things like the avatar rendering , so instead of just checking that we also check the timestamp of the item its rendering ahead of this. 
This is advantageous as its 1:1 connected to what we're drawing which is something we mostly had to keep in sync by checking the types (filtering out invisible messages to ignore in our calculations).

Additionally this handles when we load up and discover that the timestamp we previously put in there actually needs to be moved up further (like the top most message really was like 3 seconds after the new incoming messages we just loaded).

This also exposed a problem with clearing out a cache for cell heights on the desktop side, likely fixing some other 'gap' drawing issues.

### CODE REVIEW

File: shared/chat/conversation/list/index.desktop.js [hunk 1]
```diff
@@ -72,6 +72,7 @@ class BaseList extends Component<void, Props, State> {
       // Force the grid to throw away its local index based cache. There might be a lighterway to do this but
       // this seems to fix the overlap problem. The cellCache has correct values inside it but the list itself has
       // another cache from row -> style which is out of sync
+      this._cellCache.clearAll()
```
Comment (chrisnojima) [inline shared/chat/conversation/list/index.desktop.js]: else this cache can still have bad values sometimes (not always)

File: shared/chat/conversation/messages/wrapper/container.js [hunk 2]
```diff
@@ -38,6 +39,17 @@ const mapStateToProps = (state: TypedState, {messageKey, prevMessageKey}: OwnPro
   const isFirstNewMessage = !!(conversationState && message && message.messageID && conversationState.get('firstNewMessageID') === message.messageID)
   const prevMessage = getMessage(state, prevMessageKey)
   const skipMsgHeader = prevMessage && prevMessage.type === 'Text' && prevMessage.author === author
+
+  const firstMessageEver = !prevMessage
```
Comment (chrisnojima) [inline shared/chat/conversation/messages/wrapper/container.js]: we show it if its the actual first message, or the top message while we're loading, or if there's enough time between us and our neighbor

**Comment by chrisnojima** [comment]
@keybase/react-hackers

File: shared/chat/conversation/messages/wrapper/container.js [hunk 3]
```diff
@@ -38,6 +39,17 @@ const mapStateToProps = (state: TypedState, {messageKey, prevMessageKey}: OwnPro
...
+  const firstLegitMessage = prevMessage && Constants.messageKeyValue(prevMessage.key) === '1'
```
Comment (MarcoPolo) [inline shared/chat/conversation/messages/wrapper/container.js]: I don't understand why this means it's the fristLegitMessage?

File: shared/chat/conversation/messages/wrapper/container.js [hunk 4]
```diff
@@ -100,10 +114,11 @@ export default compose(
...
     componentDidUpdate: function (prevProps: Props & {_editedCount: number}) {
-      if (this.props.measure &&
+      if (this.props.measure && (
         (this.props._editedCount !== prevProps._editedCount) ||
-        (this.props.isFirstNewMessage !== prevProps.isFirstNewMessage)
-      ) {
+        (this.props.isFirstNewMessage !== prevProps.isFirstNewMessage) ||
+        (this.props.timestamp !== prevProps.timestamp)
+      )) {
         this.props.measure()
```
Comment (MarcoPolo) [inline shared/chat/conversation/messages/wrapper/container.js]: 🐑 🐑 this.props.measure && this.props.measure(). then you don't need the && ( ... || ... || ... )

double sheep - totes optional

File: shared/chat/conversation/messages/wrapper/container.js (hunk 3 above)
Comment (chrisnojima) [inline shared/chat/conversation/messages/wrapper/container.js]: renamed
